tests/ui/single-use-lifetime/issue-107998.rs

```diff
@@ -0,0 +1,18 @@
+#![deny(single_use_lifetimes)]
+#![allow(dead_code)]
+#![allow(unused_variables)]
+
+fn with<R>(f: &fn<'a>(x: &'a int) -> R) -> R {
+    //~^ ERROR function pointer types may not have generic parameters
+    //~| ERROR cannot find type `int` in this scope
```

Comment:
Is this second error needed to reproduce the ICE? Can we suppress the `int` error? Also, is the `return_it` function needed? It's best to minimize this as far as possible, I think.

Reply:
cleaned up!